### PR TITLE
fix: Unable to load persisted SMS configurations [DHIS2-14351]

### PR DESCRIPTION
**Pull request by zubaira**

### CODE REVIEW

**Review by enricocolasante — COMMENTED** [review]
I don't know very well this part and it seems that it is quite difficult to test, even with an e2e-test.
If you can't add an automated test, I would suggest to at least provide in the description or in a Jira comment a detailed step by step manual test

File: dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/sms/config/DefaultGatewayAdministrationService.java [hunk 1]
```diff
@@ -60,7 +64,12 @@
     // GatewayAdministrationService implementation
     // -------------------------------------------------------------------------
 
-    private AtomicBoolean hasGateways = null;
+    @EventListener
+    public void handleContextRefresh( ContextRefreshedEvent contextRefreshedEvent )
+    {
+        initState();
```
[the remark this answers is not in the document]
Comment (teleivo) [reply dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/sms/config/DefaultGatewayAdministrationService.java:70]: Also please add a description on how to test this manually as @enricocolasante suggested to https://dhis2.atlassian.net/browse/DHIS2-14351

**Comment by zubaira** [comment]
> I don't know very well this part and it seems that it is quite difficult to test, even with an e2e-test. If you can't add an automated test, I would suggest to at least provide in the description or in a Jira comment a detailed step by step manual test

Added the procedure to test this ticket

File: dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/sms/config/DefaultGatewayAdministrationService.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (zubaira) [reply dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/sms/config/DefaultGatewayAdministrationService.java:70]: > Could this state be initialized in a constructor? Being able to create an instance of a class that is not in a valid state is a bad design to me.

Valid point. But initState()` will be called before this class can perform any operation.

**Pull request merged**